### PR TITLE
62 refactor dashboard

### PR DESCRIPTION
**Pull request by Leepak777**
Now Dashboard has modularized into 3 subcomponents

1. Popup
-  shows title location date and description
2. MemoItem
-  display memo title and a delete button that delete memo, click component when show Popup
3. LocationItem
-  display Location name and a list of memos saved with Location

### CODE REVIEW

File: frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import MemoItem from "../MemoItem/MemoItem";
+
+const LocationItem = (props:{ 
+    locationName:string,
+    expanded:boolean,
+    handleLocationClick:(locationName:string) => void,
+    memos:[],
```
Comment (jerrydngzh) [inline frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx:7]: nit: add the appropriate type for this and other places where you use memo,

Comment (jerrydngzh) [reply frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx:7]: *avoid* `any`, might seem like a pita at first, but will make everything smoother as you know for sure the type of an object you're working with and their properties

Comment (jerrydngzh) [reply frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx:7]: case in point:
```tsx
// line 33, Popup.tsx
...
    selectedMemo.name = newTitle;
    selectedMemo.description = newMemo;
    const updatedMemo = {
      id: selectedMemo**._id**,
      description: selectedMemo.description,
      date: new Date().toLocaleString(),
...
```
- selectedMemo is a `any` type, as you never specify it, so you dont actually know if a property you access actually exists. You're just *assuming* it has it, but it doesn't
- a `MemoType`'s id is `id` not `_id`

File: frontend/src/components/Dashboard/page.tsx [hunk 2]
```diff
@@ -39,68 +23,42 @@ const Dashboard = () => {
...
-        memo: filteredMemos,
-      };
-    }
-
-    return filtered;
-  }, {});
-
-  // FIXME: this function is broken, remove `//@ts-nocheck` and fix
   const handleDeleteMemo = async (
     memo: any,
```
Comment (jerrydngzh) [inline frontend/src/components/Dashboard/page.tsx:47]: `any`

File: frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx (hunk 1 above)
Comment (jerrydngzh) [reply frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx:7]: the above non-existant property also exists several place in the code, not just here

Comment (Toastjpg) [reply frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx:7]: the _id vs id is actually a discrepency between the the MemoType interface and mongoose schema. 

The object returned from mongo will have _id but currently the type has a field of id.

A problem I outlined in #64

Comment (Toastjpg) [reply frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx:7]: So i believe the refactoring of the types of memo objects to something that isnt any will have to come in a later refactoring PR when the models are made consistent.

Comment (jerrydngzh) [reply frontend/src/components/Dashboard/components/LocationItem/LocationItem.tsx:7]: r u sewious wight neow
(very good catch, forgot about the _id type that mongoose returns)

**Review by jerrydngzh — APPROVED** [review]
lgtm, things are much cleaner now and also digestable in understanding how the dashboard components work

File: frontend/src/components/Dashboard/components/MemoItem/MemoItem.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    <div className="memo-item">
+            <div className="memo-content" onClick={(event) => handleMemoClick(event)}>
+                <span className="memo-name">{props.memo.name}</span>
+                <button className="delete-button" onClick={(event) => props.handleDeleteMemo(props.memo, event)}>Delete</button>
+            </div>
+      
+      {showPopup && (
+        <Popup
+          key={props.memo._id}
+          userID={currentUser.uid}
```
[the remark this answers is not in the document]
Comment (Leepak777) [reply frontend/src/components/Dashboard/components/MemoItem/MemoItem.tsx]: just moved it to popup in new commit

**Comment by Leepak777** [comment]
for now I'm doing this to save memo.id, will chane again after mongo schema and memo model is updated
```const getLocations = (memoList:any[]) => {
    const fetchedLocations = {};
    memoList.forEach((memo) => {
      memo.id = memo._id;
      console.log(memo);
      if (fetchedLocations[memo.location.name]) {
        fetchedLocations[memo.location.name].push(memo);
      } else {
        fetchedLocations[memo.location.name] = [memo];
      }
    });
    return fetchedLocations;
  };

**Pull request merged**